### DISCUSSION TITLE
🐞 422 Unprocessable Entity

### DISCUSSION THREAD

**Discussion by sagikazarmark** [dagger/dagger#12205] [Help]
### What is the issue?

[This](https://github.com/sagikazarmark/daggerverse/pull/177) change in my trivy module cuases the following error:

```
invoke: returned error 422 Unprocessable Entity: {"errors":[{"message":"Expected Name, found \u003cInvalid\u003e","locations":[{"line":1,"column":1677}],"extensions":{"code":"GRAPHQL_PARSE_FAILED"}}],"data":null}
```

### Dagger version

dagger v0.13.1 (registry.dagger.io/engine:v0.13.1) darwin/arm64

### Steps to reproduce

See the above PR.

### Log output

https://github.com/sagikazarmark/daggerverse/actions/runs/10963001438/job/30443530927?pr=177

**Comment by jedevc** [comment]
As mentioned in https://github.com/sagikazarmark/daggerverse/pull/177#discussion_r1771215596, this is caused by passing an invalid enum value directly to graphql.

This isn't correct, but we should make the error message clearer. Ideally, we should perform some sdk-side validation to ensure that the enum value is actually valid before attempting to marshal it into the query.

I think this is just a go issue, since other languages have stronger enum types that don't suffer from this particular issue.

**Comment by verdverm** [comment]
I have a similar error

```
returned error 422: {"data":null,"errors":[{"message":"Expected Name, found :","locations":[{"line":1,"column":104}],"extensions":{"code":"GRAPHQL_PARSE_FAILED"}}]}
exit status 1
```

I have no idea what bad data I'm passing to the Go SDK. At least getting at the actual query text could help my narrow things down. A better error message would be appreciated greatly

---

I narrowed my issue down to

```go
NetworkProtocolTcp NetworkProtocol = "TCP"
```

where I have a lowercase version and was doing the following

```go
	c = c.WithExposedPort(cfg.Port, dagger.ContainerWithExposedPortOpts{
		Description:                 cfg.Name,
		Protocol:                    dagger.NetworkProtocol(cfg.Protocol),   <=== lowercase here -> ""?
		ExperimentalSkipHealthcheck: cfg.ExperimentalSkipHealthchecks,
	})
```
